tests/api2/test_300_nfs.py

```diff
@@ -1161,3 +1161,30 @@ def test_55_checking_nfs_disable_at_boot(request):
 def test_56_destroying_smb_dataset(request):
     results = DELETE(f"/pool/dataset/id/{dataset_url}/")
     assert results.status_code == 200, results.text
+
+
+def test_60_start_nfs_service_with_empty_exports():
```

Comment:
The tests are a bit confusing to me but maybe I'm misunderstanding 😄 With your changes here, there is no reason to create an empty exports file? Simply starting the nfs service will generate an export file and the service will start, right?

Lastly, should we test the scenario of creating a file in `/etc/exports.d/` and making sure the alert is raised appropriately? Furthermore ensuring that the alert is removed if there are no files in `/etc/exports.d`?

Reply:
> Simply starting the nfs service will generate an export file and the service will start

That is correct. I'll modify the test to run both empty and missing.

Testing files actually in /etc/exports.d is a bit more complicated as our code makes it _really_ hard to actually put anything in that  directory.   I would have to create a file that 'root' is unable to remove.  This could be tricky and problematic in our automated testing.   Let me know if you know of a safe was to do this.    I did my local testing by pointing the check to a different directory.

Reply:
immutable files work and it exposed that I'm not getting a companion alert set.  
Going to place this in WIP until I get that ironed out.
Thanks, Caleb, for being detailed!